spacy/ml/tb_framework.py

```diff
@@ -139,6 +141,9 @@ def forward(model, docs_moves: Tuple[List[Doc], TransitionSystem], is_train: boo
     nO = model.get_dim("nO")
     nI = model.get_dim("nI")
 
+    beam_width = model.attrs.get("beam_width", 1)
+    beam_density = model.attrs.get("beam_density", 0.0)
```

Comment:
> These values are being defined by the beam_parser factory, right? So we should somehow pass those through?

They are set on the model in the `Parser.{predict, greedy_parse, beam_parse}` methods:

https://github.com/explosion/spaCy/blob/8cc29154acf0a3f38da06c3af010d34f4071df65/spacy/pipeline/transition_parser.pyx#L241

Though, it probably makes sense to make this configurable for `TransitionModel` as well, in case someone uses `TransitionModel` outside these methods? I have changed the `TransitionModel` constructor to accept these setttings.